readme.md

```diff
@@ -1,9 +1,224 @@
+
+A match with a bigger sequence is always better, but if we won't discard those subsequence assignments they will be always implied, which doesn't add information for the analysis. So this filter allows to reduce the size of the database while leaving only most informative assignments.
+
+After this step a BLAST database is generated from the sequences that passed the filter.
+
+Around **72%** of the sequences from `filter1` pass `filter3`. Among these sequences there are also some with a reduced number of assignments.
+
+
+### `filter3`
+
+This step actually consists of two:
+
+- Run MG7 pipeline using the reference database from `filter2` using its output sequences FASTA as input data. This will produce a table which matches each sequence the lowest common ancestor (LCA) of the assignments of any other (very) similar sequences.
+
+- For each sequence we check the relation of its assignments with the corresponding LCA that we've got from MG7. If some assignment is too far away from the LCA in the taxonomic tree, it is discarded.  
+  If a sequence has a single assignment, it is ignored, because it may represent a rare organism and it may be misplaced in the taxonomic tree.
+
+After this step BLAST database is generated again.
+
+Over **99.8%** of the sequences from `filter2` pass `filter2`, because it's mostly about reducing the number of assignments and there are not many sequences that get all assignments discarded.
+
+#### Example of a wrong assignment
+
+Let's consider the following fragment of the taxonomic tree:
+
+```
+tribeX
+├─ genusABC
+│  ├─ ...
+│  ...
+│  └─ subgenusABC
+│     ├─ speciesA
+│     │  ├─ subspeciesA1
+│     │  └─ subspeciesA2
+│     ├─ speciesB1
+│     ├─ speciesB2
+│     └─ speciesC
+...
+└─ ...
+   └─ ...
+      └─ speciesX
+```
+
+And a sequence with following taxonomic assignments:
+
+| ID   | Taxas                                |
+|:-----|:-------------------------------------|
+| seqA | subspeciesA1; subspeciesA2; speciesX |
+
+In this case `speciesX` is *likely* a wrong assignment, because it's completely *unrelated* with the other, more specific assignments. If we will leave it in the database, the LCA of these nodes will be `tribeX`, instead of `speciesA`.
+
+So we run MG7, and BLAST tells us that `seqA` is *very* similar to `seqB` and `seqC` with the following assignments:
+
+| ID   | Taxas                |
+|:-----|:---------------------|
+| seqB | speciesB1; speciesB2 |
+| seqC | speciesC             |
+
+We take their LCA which is `subgenusABC` and look at its parent: `genusABC`. Each of the `seqA`'s assignments has to be a descendant of `genusABC` and `speciesX`, obviously, is not, so we discard it:
+
+| ID   | Taxas                      |
+|:-----|:---------------------------|
+| seqA | subspeciesA1; subspeciesA2 |
+
```

Comment:
This is fine. It needs a bit of polishing, but will serve as a starting point for the relevant paper section.